### PR TITLE
Add setup environment guide to README

### PR DESCRIPTION
**Pull request by cfcosta**
[skip ci]
This should simplify the lifes of anyone wanting to contribute.

### CODE REVIEW

**Comment by cfcosta** [comment]
@abonas regarding your comments on the gist:

>why running bin/update is necessary on all repos ? not only on the core repo?

Right now it updates the `spec/manageiq` clone of the repo, so it's necessary. I think it might be worth adding a section on the guide to also guide on how to symlink the manageiq used on tests to the same being used in development, what do you think?

>adding a link to the manageiq developer guide is misleading as it describes a different method of work with the other repos (via the plugins folder) - it can be confusing for newcomers that might attempt to do both ways

That's true, should I copy the environment setup part to our side as well?

**Comment by abonas** [comment]
> That's true, should I copy the environment setup part to our side as well?

that will require us to maintain this doc in addition to the main guide which can be a burden

File: README.md [hunk 1]
```diff
@@ -14,9 +14,348 @@ ManageIQ plugin for the Hawkular provider.
...
+
+It will ask then some questions:
+
+* What should it use for running the services? Docker-Compose? Openshift?
+* What version of Hawkular?
+* What version of Cassandra?
+* What WildFly version?
+* Standalone or Domain mode for Wildfly?
+* How many WildFly servers?
+
```
Comment (abonas) [inline README.md]: this is great. I think currently missing:
1. information on configuring mutable vs immutable servers - as in miq we have functionality specific to each case
2. how to start an additional server on an already running env

File: README.md [hunk 2]
```diff
@@ -14,9 +14,345 @@ ManageIQ plugin for the Hawkular provider.
...
+For OS X, `rbenv` and `ruby-build` are available on homebrew, and can be installed like this:
+
+```bash
+brew upgrade rbenv ruby-build
+echo 'eval "$(rbenv init -)"' >> ~/.bashrc
+eval "$(rbenv init -)"
+```
+
+For Linux, it will have to be installed via default git checkout:
```
Comment (abonas) [inline README.md]: isn't it easier with rvm?

File: README.md [hunk 3]
```diff
@@ -14,9 +14,345 @@ ManageIQ plugin for the Hawkular provider.
...
+
+### Updating the repositories
+
+To update the repositories, you need to go on each one and run `bin/update`. This script takes care of it:
+
+```bash
+#!/bin/bash
+
+MIQ_DIR="$HOME/manageiq"
+OLD_PWD="$(pwd)"
```
Comment (abonas) [inline README.md]: what is this for?

File: README.md [hunk 4]
```diff
@@ -14,9 +14,345 @@ ManageIQ plugin for the Hawkular provider.
...
+git remote add upstream git@github.com:ManageIQ/manageiq-ui-classic
+cd ..
+
+cd manageiq-schema
+git remote add upstream git@github.com:ManageIQ/manageiq-schema
+cd ..
+
+cd manageiq-api
+git remote add upstream git@github.com:ManageIQ/manageiq-api
+cd ..
```
Comment (abonas) [inline README.md]: can't all cd point to the next folder already without first going to ".." ?

e.g. 
 +cd manageiq-api
 +git remote add upstream git@github.com:ManageIQ/manageiq-api
 +cd ../manageiq-providers-hawkular

File: README.md [hunk 5]
```diff
@@ -14,9 +14,345 @@ ManageIQ plugin for the Hawkular provider.
...
+Any Linux/OS X System is compatible, as far as we are concerned.
+
+### Installing Dependencies
+
+#### Fedora
+
+First, install all system dependencies:
+
+```bash
+sudo dnf -y git-all memcached postgresql-devel postgresql-server postgresql-contrib bzip2 libffi-devel readline-devel sqlite-devel nodejs gcc-c++ libcurl-devel npm openssl-devel cmake openscap
```
Comment (abonas) [inline README.md]: docker missing here for install?

Comment (josejulio) [reply README.md]: Isn't this all (or part of) already here? https://github.com/ManageIQ/guides/blob/master/developer_setup.md

File: README.md [hunk 6]
```diff
@@ -14,9 +14,345 @@ ManageIQ plugin for the Hawkular provider.
...
+
+There are two ways of installing Hawkular: manually or via Docker. There are
+use cases for both, and you can mix and match for your needs, but the simplest
+way of getting it working is via Hawkinit, which downloads, configures and runs
+the hawkular server, and 0+ wildfly instances with the Hawkular agent installed
+and configured.
+
+#### Via Docker (recommended)
+
+First, we need to install [hawkinit](https://github.com/Jiri-Kremser/hawkinit),
```
Comment (abonas) [inline README.md]: isn't this on https://github.com/hawkular/hawkinit ?

Comment (josejulio) [reply README.md]: Yes, we should link there and provide hawkinit recipes for common scenarios.

File: README.md [hunk 7]
```diff
@@ -14,9 +14,345 @@ ManageIQ plugin for the Hawkular provider.
...
+`docker` group:
+
+```bash
+sudo usermod -a -G docker `whoami`
+newgrp docker # to update your groups on the session
+```
+
+Then, by running `hawkinit`, you'll see something like this:
+
+![](screenshots/hawkinit.png)
```
Comment (abonas) [inline README.md]: screenshot needs fixing after [this](https://github.com/hawkular/hawkinit/pull/38) will be merged

File: README.md (hunk 5 above)
Comment (abonas) [reply README.md]: yes, I commented about it [here](https://github.com/ManageIQ/manageiq-providers-hawkular/pull/38#issuecomment-327071406)

File: README.md (hunk 2 above)
Comment (josejulio) [reply README.md]: I'm not sure, I think is matter of choice. I use `chruby`. 
If rvm setup is easier we should provide that one here instead, unless there is a real advantage of using rbenv.

File: README.md (hunk 1 above)
Comment (josejulio) [reply README.md]: I added some info here: https://github.com/cfcosta/manageiq-providers-hawkular/pull/1

File: README.md (hunk 2 above)
Comment (abonas) [reply README.md]: @cfcosta ^^

File: README.md (hunk 4 above)
Comment (abonas) [reply README.md]: @cfcosta ^^

File: README.md (hunk 5 above)
Comment (abonas) [reply README.md]: @josejulio but docker is missing in that dev guide because it is relevant for hawkinit but not for miq at the moment

File: README.md (hunk 7 above)
Comment (abonas) [reply README.md]: @cfcosta [this](https://github.com/hawkular/hawkinit/pull/38) has been merged, please update the screenshot

**Comment by abonas** [comment]
please add [skip ci] in the top comment

File: README.md (hunk 5 above)
Comment (abonas) [reply README.md]: maybe it's still worth to have a guide of our own because we can update it as frequent as we want and make it tailored for our needs

File: README.md [hunk 8]
```diff
@@ -14,9 +14,345 @@ ManageIQ plugin for the Hawkular provider.
 
 ## Development
 
-See the section on pluggable providers in the [ManageIQ Developer Setup](http://manageiq.org/docs/guides/developer_setup)
+This is a quick guide outlining all the necessary steps to get ManageIQ up and
```
Comment (israel-hdez) [inline README.md]: I have the idea that this doesn't fit well in the `README.md`. This is converting the readme file into a developer quick start, because readme doesn't have a lot of info. I think it's better to have this guide in its own file and place a link here.

File: README.md (hunk 4 above)
Comment (israel-hdez) [reply README.md]: Instead of cd-ing, you could also pass -C flag to simplify:

`git -C manageiq-ui-classic/  remote add upstream git@github.com:ManageIQ/manageiq-ui-classic`

File: README.md [hunk 9]
```diff
@@ -14,9 +14,345 @@ ManageIQ plugin for the Hawkular provider.
...
+```ruby
+gem 'pry'
+gem 'pry-byebug'
+```
+
+The next step is to configure each repository:
+
+```bash
+cd ~/ManageIQ/manageiq
+./bin/setup
```
Comment (israel-hdez) [inline README.md]: isn't enough to run only this `bin/setup`? The other ones aren't needed unless you are running tests. But in order to run tests, some extra setup is required.

File: README.md [hunk 10]
```diff
@@ -14,9 +14,345 @@ ManageIQ plugin for the Hawkular provider.
...
+
+cd hawkular-client-ruby
+bundle install
+cd ..
+```
+
+This should be enough. To run the application, we can run this, which will run the application on background on port 3000:
+
+```bash
+bundle exec rake evm:start
```
Comment (israel-hdez) [inline README.md]: You should be in `manageiq` directory (main repo).

File: README.md [hunk 11]
```diff
@@ -14,9 +14,345 @@ ManageIQ plugin for the Hawkular provider.
...
+Sign in with the default credentials (`admin`/`smartvm`), and you should be good to go!
+
+### Updating the repositories
+
+To update the repositories, you need to go on each one and run `bin/update`. This script takes care of it:
+
+```bash
+#!/bin/bash
+
+MIQ_DIR="$HOME/manageiq"
```
Comment (israel-hdez) [inline README.md]: This should be `MIQ_DIR="$HOME/ManageIQ"` to be consistent with directory name at [line 138](https://github.com/ManageIQ/manageiq-providers-hawkular/pull/38/files#diff-04c6e90faac2675aa89e2176d2eec7d8R138).

File: README.md [hunk 12]
```diff
@@ -14,9 +14,345 @@ ManageIQ plugin for the Hawkular provider.
...
+    git fetch -a
+    git pull --rebase upstream master
+    ./bin/update
+    cd -
+  fi
+done
+
+echo "Updating core..."
+
+cd $MIQ_DIR/core
```
Comment (israel-hdez) [inline README.md]: This should be `cd $MIQ_DIR/manageiq` to be consistent with name used at [line 127](https://github.com/ManageIQ/manageiq-providers-hawkular/pull/38/files#diff-04c6e90faac2675aa89e2176d2eec7d8R127)

**Comment by cfcosta** [comment]
@abonas updated.

File: README.md (hunk 8 above)
Comment (abonas) [reply README.md]: sure, we can move this to another file.

**Comment by abonas** [comment]
@cfcosta are you sure you pushed the changes? I don't see a lot of questions/comments addressed.
for example: the missing docker or regarding rvm or moving this to another file or changing the screenshot to a newer one where typo was fixed

File: app/models/manageiq/providers/hawkular/middleware_manager/event_catcher/runner.rb [hunk 13]
```diff
@@ -42,13 +42,13 @@ def monitor_events
       event_monitor_running
 
       # Separate alerts from avail updates
-      avail_updates, events = events.partition { |e| !e.kind_of?(::Hawkular::Alerts::Alert) }
+      events, avail_updates = events.partition { |e| !e.kind_of?(::Hawkular::Alerts::Event) }
```
Comment (josejulio) [inline app/models/manageiq/providers/hawkular/middleware_manager/event_catcher/runner.rb]: Is this file needed for the README.md in any way or was pushed by mistake?

**Comment by josejulio** [comment]
Hawkinit changes on the Readme seems to be overwritten (git push -f?)
Can you please cherry pick https://github.com/cfcosta/manageiq-providers-hawkular/pull/1/commits/0e46f0fd2760198f97192cae7222838ebb1f4411 ?

File: README.md (hunk 2 above)
Comment (cfcosta) [reply README.md]: There is. Afaik, if that didn't change, rvm overwrites system commands like cd and is generally a pain to work with, requiring implodes every now and then. I've been using rbenv since and the workflow couldn't be nicer.

Either way, I'm going to remove installation instructions and let everyone chooses what they want to use.

File: README.md (hunk 5 above)
Comment (cfcosta) [reply README.md]: Also added a guide to Docker.

File: README.md (hunk 7 above)
Comment (cfcosta) [reply README.md]: I just installed and the UI is the same, am I missing something?

File: README.md (hunk 8 above)
Comment (cfcosta) [reply README.md]: I disagree, having ManageIQ working correctly is essential to test/developer on our provider code.

File: README.md (hunk 9 above)
Comment (cfcosta) [reply README.md]: What kind of extra setup?

File: app/models/manageiq/providers/hawkular/middleware_manager/event_catcher/runner.rb (hunk 13 above)
Comment (cfcosta) [reply app/models/manageiq/providers/hawkular/middleware_manager/event_catcher/runner.rb]: Pushed by mistake.

File: README.md (hunk 7 above)
Comment (josejulio) [reply README.md]: Yes, see my commit. hawkinit is under hawkular organization now.

**Comment by cfcosta** [comment]
@abonas rebased and added a Docker guide as well, can you check if there's anything else missing?

File: README.md (hunk 8 above)
Comment (josejulio) [reply README.md]: I suppose they mean to have it more organized like this: https://github.com/ManageIQ/manageiq/blob/master/README.md

**Comment by abonas** [comment]
it should be skip ci, not ci skip :)

**Comment by abonas** [comment]
@cfcosta seems that @josejulio changes did not get in? please check
and also see the suggestion to link to this file from readme instead of it being *the* readme

File: README.md (hunk 9 above)
Comment (israel-hdez) [reply README.md]: This should be: http://www.manageiq.org/docs/guides/developer_setup/plugins#running-tests

File: README.md [hunk 14]
```diff
@@ -14,9 +14,336 @@ ManageIQ plugin for the Hawkular provider.
...
+cd $MIQ_DIR/manageiq
+
+./bin/update
+
+cd $OLD_PWD
+
+echo "DONE."
+```
+
+## Installing Docker
```
Comment (israel-hdez) [inline README.md]: I'm not very, very sure because I don't remember well... But IIRC, docker-compose is installed separately from the engine. That should be mentioned here because hawkinit depends on compose.

Comment (cfcosta) [reply README.md]: That's true, will add it.

**Comment by mtho11** [comment]
This getting hard to follow with all the comments now. Let's merge and get a more contiguous version. But keep up with the commentary.

File: README.md (hunk 9 above)
Comment (cfcosta) [reply README.md]: Afaik the `spec:setup` task is also run when you run `bin/setup`, so we should be covered there.

**Comment by mtho11** [comment]
@abonas I think we should merge what we have and keep adding to it.

**Comment by abonas** [comment]
merging. I see currently some work to be done from the comments above:
1. fix screenshot of hawkinit to a new one without typo
2. move this to another page and link from readme
3. @josejulio is your contribution to @cfcosta fork here or still ovewritten? (that's why I originally commented it's better to put it directly here to avoid overrides :))

**Comment by josejulio** [comment]
> merging. I see currently some work to be done from the comments above:
> 
> fix screenshot of hawkinit to a new one without typo - this issue is tracking it
> @josejulio is your contribution to @cfcosta fork here or still ovewritten? (that's why I originally commented it's better to put it directly here to avoid overrides :))

Is here.

> @josejulio is there a way to document commands how to add a new additional eap server to the documented setup?

Sure, will add the info
